sonar/modules/documents/marshmallow/json.py

```diff
@@ -34,6 +34,32 @@
                                schema=DocumentRecord.schema)
 
 
+class FileSchemaV1(StrictKeysMixin):
+    """File schema."""
+
+    bucket = SanitizedUnicode()
+    file_id = SanitizedUnicode()
+    version_id = SanitizedUnicode()
+    key = SanitizedUnicode()
+    checksum = SanitizedUnicode()
+    size = fields.Number()
+    label = SanitizedUnicode()
+    type = SanitizedUnicode()
+    order = fields.Number()
+    external_url = SanitizedUnicode()
+    restricted = SanitizedUnicode()
+    embargo_date = SanitizedUnicode()
+    restriction = fields.Dict(dump_only=True)
+
+    @pre_load
+    def remove_restriction(self, data):
+        """Remove restriction information before saving."""
+        if data.get('restriction'):
+            data.pop('restriction')
```

Comment:
You could use `data.pop('restriction', None)` to not raised an KeyError if 'restriction' key doesn't exists.

Reply:
Thanks for the trick!